### ISSUE TITLE
AOP-driven Toggling

### ISSUE THREAD

**Issue by leandroleo02** [clun/ff4j#285]
Hi!

I'm using the **"AOP-driven Toggling"** and it works pretty well when I have two features and I want to use one or another. 

But now, I'm creating a completely new feature and I don't have a "old one" to turn on or off.
Is it possible to use **"AOP-driven Toggling strategy"** in this case?

One workaround is create temporarily a dummy object just to do nothing, but it doesn't seems to good.

Any help?

Thanks in advance! :-)

**Comment by 7fantasy7** [comment]
+1 Interested in this possibility too:)

**Comment by clun** [comment]
Hi,

I think i understood what you mean. 

Today the `Flip` annotation allows to switch from one implementation to another. As long as there are 2 implementations everything is fine. But when creating new method you do not want to create an empty method aside just for the case where feature is disabled (ad providing `alterBean` or `alterClass` for nothing).

```java
@Flip(name = "featureX", alterBean = "greeting.french")
public String myNewFeatureX(String name);
```

You would like simply to "do nothing" when the feature is off. It might indeed be a good idea (and quite easy, have to return null [HERE](https://github.com/clun/ff4j/blob/master/ff4j-aop/src/main/java/org/ff4j/aop/FeatureAdvisor.java#L63) if no alter bean or alter class provided ). 

Still, if the method is expected an object **I will return null and as a consequence we have to be carefull**. Would you be ok with such implementation ?

*(v2.X is in heavy development with Java8 support, Optional would have been good here)*

**Comment by clun** [comment]
I implemented this small evolution.

As the implementation is really based on Interface. You will still need to define an Interface for your new feature even if a single implementation. Will update this behavior in 2.x.

```java
public interface SampleService {
	@Flip(name = "AwesomeFeature")
	String someAwesomeFeature(String param1);
}

@Component("ff4j.aop.noalterbean")
public class SampleServiceFeature implements SampleService {
	public String someAwesomeFeature(String param1) { return param1;}
}

// [...]
@Autowired
private FF4j ff4j;

@Autowired
private SampleService sample;

@Test
public void testAOP() {
    	// AwesomeFeature exist and is enabled
    	Assert.assertEquals("HELLO", sample.someAwesomeFeature("HELLO");
    	
    	ff4j.disable("AwesomeFeature");
    	
    	// Then
    	Assert.assertNull(sample.someAwesomeFeature("HELLO"));
    }
```

packaging in progress....

**Comment by clun** [comment]
released 1.7.1 !
